vendor/diff-parsers-upstream-std.patch

```diff
@@ -0,0 +1,821 @@
+     | Pexp_extension x -> extension ~loc ~attrs (sub.extension sub x)
+     | Pexp_unreachable -> unreachable ~loc ~attrs ()
++    | Pexp_hole -> hole ~loc ~attrs ()
+ 
+   let map_binding_op sub {pbop_op; pbop_pat; pbop_exp; pbop_loc} =
+     let open Exp in
+     let op = map_loc sub pbop_op in
+     let pat = sub.pat sub pbop_pat in
+@@ -697,13 +701,15 @@
+            ~attrs:(this.attributes this pvb_attributes)
+       );
+ 
+ 
+     constructor_declaration =
+-      (fun this {pcd_name; pcd_args; pcd_res; pcd_loc; pcd_attributes} ->
++      (fun this {pcd_name; pcd_vars; pcd_args;
++                 pcd_res; pcd_loc; pcd_attributes} ->
+         Type.constructor
+           (map_loc this pcd_name)
++          ~vars:(List.map (map_loc this) pcd_vars)
+           ~args:(T.map_constructor_arguments this pcd_args)
+           ?res:(map_opt (this.typ this) pcd_res)
+           ~loc:(this.location this pcd_loc)
+           ~attrs:(this.attributes this pcd_attributes)
+       );
+--- ocaml-4.13-upstream/parser.mly
++++ ocaml-4.13/parser.mly
+@@ -206,21 +206,21 @@
+ let mkstrexp e attrs =
+   { pstr_desc = Pstr_eval (e, attrs); pstr_loc = e.pexp_loc }
+ 
+ let mkexp_constraint ~loc e (t1, t2) =
+   match t1, t2 with
+-  | Some t, None -> ghexp ~loc (Pexp_constraint(e, t))
+-  | _, Some t -> ghexp ~loc (Pexp_coerce(e, t1, t))
++  | Some t, None -> mkexp ~loc (Pexp_constraint(e, t))
++  | _, Some t -> mkexp ~loc (Pexp_coerce(e, t1, t))
+   | None, None -> assert false
+ 
+ let mkexp_opt_constraint ~loc e = function
+   | None -> e
+   | Some constraint_ -> mkexp_constraint ~loc e constraint_
+ 
+ let mkpat_opt_constraint ~loc p = function
+   | None -> p
+-  | Some typ -> ghpat ~loc (Ppat_constraint(p, typ))
++  | Some typ -> mkpat ~loc (Ppat_constraint(p, typ))
+ 
+ let syntax_error () =
+   raise Syntaxerr.Escape_error
+ 
+ let unclosed opening_name opening_loc closing_name closing_loc =
+@@ -387,16 +387,16 @@
+   loc_map Longident.last id
+ 
+ let loc_lident (id : string Location.loc) : Longident.t Location.loc =
+   loc_map (fun x -> Lident x) id
+ 
+-let exp_of_longident ~loc lid =
+-  let lid = make_ghost (loc_map (fun id -> Lident (Longident.last id)) lid) in
+-  ghexp ~loc (Pexp_ident lid)
++let exp_of_longident lid =
++  let lid = loc_map (fun id -> Lident (Longident.last id)) lid in
++  Exp.mk ~loc:lid.loc (Pexp_ident lid)
+ 
+-let exp_of_label ~loc lbl =
+-  mkexp ~loc (Pexp_ident (loc_lident lbl))
++let exp_of_label lbl =
++  Exp.mk ~loc:lbl.loc (Pexp_ident (loc_lident lbl))
+ 
+ let pat_of_label lbl =
+   Pat.mk ~loc:lbl.loc  (Ppat_var (loc_last lbl))
+ 
+ let mk_newtypes ~loc newtypes exp =
+@@ -822,13 +822,13 @@
+ %nonassoc HASH                         /* simple_expr/toplevel_directive */
+ %left     HASHOP
+ %nonassoc below_DOT
+ %nonassoc DOT DOTOP
+ /* Finally, the first tokens of simple_expr are above everything else. */
+-%nonassoc BACKQUOTE BANG BEGIN CHAR FALSE FLOAT INT
```

Comment:
Why do we add OBJECT here ?

Reply:
This is due to https://github.com/ocaml-ppx/ocamlformat/pull/1755, maybe we need to compare with the `trunk` branch instead of `4.13`

Reply:
Well trunk is a moving target. What about a specific commit that we would bump from time to time ?
Also if we're already supporting stuff from after 4.13, we should maybe not name directories with 4.13 in it.

Reply:
I updated the version used for the vendored upstream, I agree that "4.13" is not exact, that's some kind of 4.13+, but not yet 4.14 and it might still change before 4.14. Personally it does not confuse me (4.13 is more informative than ocaml_current or ocaml_trunk) we can change it in another PR if we find a better name, since there are more impact in the code.